Formula/g/glaze.rb

```diff
@@ -0,0 +1,61 @@
+class Glaze < Formula
+  desc "Extremely fast, in-memory JSON and interface library for modern C++"
+  homepage "https://github.com/stephenberry/glaze"
+  url "https://github.com/stephenberry/glaze/archive/refs/tags/v4.1.0.tar.gz"
+  sha256 "2cb3e650a45738f7e7e67e35683ab0b91b7be1cf42f819f33f576ad86871a1fb"
+  license "MIT"
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "llvm" => [:build, :test]
+
+  fails_with :gcc do
+    version "11"
+    cause "Requires C++20"
+  end
+
+  def install
+    # Issue ref: https://github.com/stephenberry/glaze/issues/1500
+    ENV.append_to_cflags "-stdlib=libc++" if OS.linux?
```

Comment:
Note that in general, this is not acceptable, because it means LLVM's libc++ becomes its runtime dependency and needs to be installed alongside the formula. And that's way too heavy.

However, Glaze is a header-only library which means we shouldn't need to build anything. In fact the C++ executable built here, as seen in the logs in https://github.com/stephenberry/glaze/issues/1500, is a ["developer mode" thing](https://github.com/stephenberry/glaze/blob/v4.2.1/cmake/dev-mode.cmake) and not even installed. So we can avoid building it at all by passing `-Dglaze_DEVELOPER_MODE=OFF`. That also eliminates the need for a LLVM build/runtime dependency.

It's OK to test the formula using LLVM, though, since it's header-only and has no additional dependencies which means it should work for any compiler with a decent level of C++20 support.

Reply:
Great analysis and very helpful. Thank you! @ZhongRuoyu 👍